tool/lib/test/unit.rb

```diff
@@ -854,6 +854,38 @@ def record(suite, method, assertions, time, error)
           end
         end
         # (((@record ||= {})[suite] ||= {})[method]) = [assertions, time, error]
+        if writer = @options[:launchable_test_reports]
+          location = suite.instance_method(method).source_location
+          if location && path = location.first
+            # Launchable JSON schema is defined at
+            # https://github.com/search?q=repo%3Alaunchableinc%2Fcli+https%3A%2F%2Flaunchableinc.com%2Fschema%2FRecordTestInput&type=code.
+            e = case error
+                when nil
+                  status = 'TEST_PASSED'
+                  nil
+                when Test::Unit::PendedError
+                  status = 'TEST_SKIPPED'
+                  "Skipped:\n#{klass}##{meth} [#{location e}]:\n#{e.message}\n"
+                when Test::Unit::AssertionFailedError
+                  status = 'TEST_FAILED'
+                  "Failure:\n#{klass}##{meth} [#{location e}]:\n#{e.message}\n"
+                when Timeout::Error
+                  status = 'TEST_FAILED'
+                  "Timeout:\n#{klass}##{meth}\n"
+                else
+                  status = 'TEST_FAILED'
+                  bt = Test::filter_backtrace(e.backtrace).join "\n    "
+                  "Error:\n#{klass}##{meth}:\n#{e.class}: #{e.message.b}\n    #{bt}\n"
```

Comment:
What do you expect as `e` here?
This variable seems unassigned yet inside `case error` block.
Maybe intended `error`?

Reply:
> Maybe intended error?
Yes that's right.

Reply:
I've addressed the change at https://github.com/ruby/ruby/pull/9777/files#diff-d58358cd56895151702a7ae7aae9bc0891f2e692a147a46f388e3abb784e354bR1378.

Thank you for pointing it out!